script/lint.js

```diff
@@ -200,17 +235,21 @@ async function main () {
   const opts = parseCommandLine()
 
   // no mode specified? run 'em all
-  if (!opts['c++'] && !opts.javascript && !opts.python && !opts.gn) {
-    opts['c++'] = opts.javascript = opts.python = opts.gn = true
+  if (!opts['c++'] && !opts.javascript && !opts.python && !opts['gn-format']) {
+    opts['c++'] = opts.javascript = opts.python = opts['gn-format'] = true
   }
 
   const linters = LINTERS.filter(x => opts[x.key])
 
   for (const linter of linters) {
-    const filenames = await findFiles(opts, linter)
-    if (filenames.length) {
-      if (opts.verbose) { console.log(`linting ${filenames.length} ${linter.key} ${filenames.length === 1 ? 'file' : 'files'}`) }
-      linter.run(opts, filenames)
+    if (linter.key === 'gn-check') {
+      linter.run(opts)
```

Comment:
we should only run `gn check` if a *.{gn,gni,h,c,cc,m,mm} file changes.

Reply:
Adding this to `lint-staged` hook should take care of it.

Reply:
since this is running in CI now, this is moot.